libraries/chain/controller.cpp

```diff
@@ -118,15 +118,16 @@ struct building_block {
    optional<producer_authority_schedule> _new_pending_producer_schedule;
    vector<digest_type>                   _new_protocol_feature_activations;
    size_t                                _num_new_protocol_features_that_have_activated = 0;
-   vector<transaction_metadata_ptr>      _pending_trx_metas;
-   vector<transaction_receipt>           _pending_trx_receipts;
-   vector<digest_type>                   _action_receipt_digests;
+   deque<transaction_metadata_ptr>       _pending_trx_metas;
+   deque<transaction_receipt>            _pending_trx_receipts; // boost deque in 1.71 with 1024 elements performs better
+   deque<digest_type>                    _pending_trx_receipt_digests;
```

Comment:
Make sure to reset the size of this new container in `make_block_restore_point`.

Reply:
Added.